source/common/http/http2/codec_impl.h

```diff
@@ -494,6 +527,7 @@ class ClientConnectionImpl : public ClientConnection, public ConnectionImpl {
   bool checkInboundFrameLimits() override { return true; }
 
   Http::ConnectionCallbacks& callbacks_;
+  Nghttp2SessionFactoryPtr http2_session_factory_;
```

Comment:
Do we need to actually store this?

Reply:
This is no longer applicable, but the goal here was to allow factories to manage state that needed to have the same lifetime as the `nghttp2_session`.